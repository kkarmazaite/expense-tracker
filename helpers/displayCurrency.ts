export const displayCurrency = (value:number) => {
  return new Intl.NumberFormat('de-DE', { 
    style: 'currency', 
    currency: 'EUR',
    currencyDisplay:'code',
  }).format(value)
}