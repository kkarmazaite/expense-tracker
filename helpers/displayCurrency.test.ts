import { expect, describe, it } from 'vitest'
import { displayCurrency } from './displayCurrency'

describe('Helper | displayCurrency', async() => {
  it('Check displayCurrency function', () => {
    const input:number = 20
    const output:string = '20,00 EUR'

    expect(displayCurrency(input).replace(/\s/, ' ')).toEqual(output);
    expect(displayCurrency(input).replace(/\s/, ' ')).toMatchSnapshot()
  })
})