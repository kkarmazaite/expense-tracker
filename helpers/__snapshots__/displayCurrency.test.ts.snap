// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Helper | displayCurrency > Check displayCurrency function 1`] = `"20,00 EUR"`;
